deployment/ccip/changeset/cs_chain_contracts.go

```diff
@@ -351,6 +353,283 @@ func UpdateOnRampsDestsChangeset(e deployment.Environment, cfg UpdateOnRampDests
 	}}, nil
 }
 
+type UpdateOnRampDynamicConfig struct {
+	// UpdatesByChain is a mapping of source -> update.
+	UpdatesByChain map[uint64]OnRampDynamicConfigUpdate
+	// Disallow mixing MCMS/non-MCMS per chain for simplicity.
+	// (can still be achieved by calling this function multiple times)
+	MCMS *MCMSConfig
+}
+
+type OnRampDynamicConfigUpdate struct {
+	FeeQuoter          common.Address
+	MessageInterceptor common.Address
+	FeeAggregator      common.Address
+	AllowlistAdmin     common.Address
+}
+
+func (cfg UpdateOnRampDynamicConfig) Validate() error {
```

Comment:
Can we move this to be directly under `UpdateOnRampDynamicConfig` for better readability?